### PR TITLE
feat: Migrate compute target TCP proxy(global) to direct controller

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Prerequisite for b/246759060

Migrate global compute target TCP proxy to direct controller. Will support regional in a followup PR.

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp-direct/_http.log [hunk 1]
```diff
@@ -267,10 +274,46 @@ X-Xss-Protection: 0
...
   "startTime": "2024-04-01T12:34:56.123456Z",
-  "status": "DONE"
+  "status": "RUNNING",
+  "targetId": "2860175380157437550",
+  "targetLink": "https://www.googleapis.com/compute/v1/projects/${projectId}/global/healthChecks/computehealthcheck-${uniqueId}",
+  "user": "user@example.com"
+}
+
+---
+
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp-direct/_http.log:288]: Looks like the real GCP using the TF-based controller. Could you also upload the one using direct controller? (please do that in a new git-commit)

File: apis/compute/v1beta1/targettcpproxy_reference.go
Comment (yuwenma) [inline apis/compute/v1beta1/targettcpproxy_reference.go:1]: suggest to start developing a shared functionality for compute resource. It looks many compute resource would need this global/regional handling.

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp-direct/_http.log (hunk 1 above)
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp-direct/_http.log]: This log is from ComputeHealthCheck and it's still using TF controller.

**Comment by yuwenma** [comment]
/lgtm
/approve

One comment about code refactoring. Otherwise looks good

File: apis/compute/v1beta1/targettcpproxy_reference.go
Comment (gemmahou) [reply apis/compute/v1beta1/targettcpproxy_reference.go:1]: Will address this in b/379723623

File: pkg/controller/direct/compute/targettcpproxy/refs.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/compute/v1beta1"
+
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ResolveComputeBackendService(ctx context.Context, reader client.Reader, src client.Object, ref *refs.ComputeBackendServiceRef) (*refs.ComputeBackendServiceRef, error) {
```
[the remark this answers is not in the document]
Comment (gemmahou) [reply pkg/controller/direct/compute/targettcpproxy/refs.go:33]: Will address this in b/379722657

**Pull request merged**